core/src/sumeragi/network_topology.rs

```diff
@@ -105,15 +108,15 @@ impl Topology {
             .cloned()
             .collect()
     }
+
     /// What role does this peer have in the topology. If it is not in the toplogy
     /// the function will return `Role::ObservingPeer`.
     // This lint is a bad suggestion.
-    #[allow(clippy::option_if_let_else)]
     pub fn role(&self, peer_id: &PeerId) -> Role {
         match self.sorted_peers.iter().position(|p| p == peer_id) {
             Some(index) if index == 0 => Role::Leader,
-            Some(index) if index < self.min_votes_for_commit() => Role::ValidatingPeer,
-            Some(index) if index == self.min_votes_for_commit() => Role::ProxyTail,
+            Some(index) if index < self.min_votes_for_commit() - 1 => Role::ValidatingPeer,
+            Some(index) if index == self.min_votes_for_commit() - 1 => Role::ProxyTail,
```

Comment:
Comment on why we need to do `-1`. This can save searching far and wide to find the reason.

Reply:
Or add self-explanatory local (or non-local) constant:

```rust
const SIZE_AND_INDEX_DIFFERENCE = 1;
```

It will be compiled away anyway (haven't checked though).

Reply:
I moved this logic into `proxy_tail_index` private function with required clarifications.